js/theme_creator.js

```diff
@@ -0,0 +1,262 @@
+    }
+  };
+
+  const THEME_DEFAULTS = {
+    default: {
+      background: '#ffffff',
+      textColor: '#4d4d4d',
+      highlightColor: '#00caf2',
+      linkColor: '#00caf2',
+      borderColor: '#e7e7e7',
+      buttonBackground: '#f4f4f4',
+      inputBackground: '#ffffff',
+      inputColor: '#333333',
+      inputClearBackground: '#909ca7',
+      headerBackground: '#ffffff',
+      headerColor: '#4d4d4d',
+      headerIconColor: '#4d4d4d',
+      headerButtonColor: '#00caf2',
+      headerDisabledButtonColor: '#e7e7e7',
+      headerActionButtonColor: '#4d4d4d'
+    },
+    communications: {
+      background: '#f4f4f4',
+      textColor: '#27c8c2',
+      highlightColor: '#00caf2',
+      linkColor: '#177874',
+      borderColor: '#e7e7e7',
+      buttonBackground: '#f4f4f4',
+      inputBackground: '#ffffff',
+      inputColor: '#333333',
+      inputClearBackground: '#909ca7',
+      headerBackground: '#27c8c2',
+      headerColor: '#ffffff',
+      headerIconColor: '#ffffff',
+      headerButtonColor: '#177874',
+      headerDisabledButtonColor: '#e7e7e7',
+      headerActionButtonColor: '#4d4d4d'
+    },
+    media: {
+      background: '#333333',
+      textColor: '#ffffff',
+      highlightColor: '#00caf2',
+      linkColor: '#00caf2',
+      borderColor: '#4d4d4d',
+      buttonBackground: '4d4d4d',
+      inputBackground: '#4d4d4d',
+      inputColor: '#2b2b2b',
+      inputClearBackground: '#909ca7',
+      headerBackground: '#333333',
+      headerColor: '#ffffff',
+      headerIconColor: '#ffffff',
+      headerButtonColor: '#00caf2',
+      headerDisabledButtonColor: '#e7e7e7',
+      headerActionButtonColor: '#4d4d4d'
+    },
+    productivity: {
+      background: '#ffffff',
+      textColor: '#4d4d4d',
+      highlightColor: '#27c8c2',
+      linkColor: '#00caf2',
+      borderColor: '#e7e7e7',
+      buttonBackground: '#f4f4f4',
+      inputBackground: '#ffffff',
+      inputColor: '#333333',
+      inputClearBackground: '#909ca7',
+      headerBackground: '#ff9900',
+      headerColor: '#ffffff',
+      headerIconColor: '#4d4d4d',
+      headerButtonColor: '#00caf2',
+      headerDisabledButtonColor: '#e7e7e7',
+      headerActionButtonColor: '#4d4d4d'
+    },
+    settings: {
+      background: '#f4f4f4',
+      textColor: '#4d4d4d',
+      highlightColor: '#00caf2',
+      linkColor: '#00caf2',
+      borderColor: '#e7e7e7',
+      buttonBackground: '#ffffff',
+      inputBackground: '#ffffff',
+      inputColor: '#333333',
+      inputClearBackground: '#909ca7',
+      headerBackground: '#f4f4f4',
+      headerColor: '#858585',
+      headerIconColor: '#4d4d4d',
+      headerButtonColor: '#00caf2',
+      headerDisabledButtonColor: '#e7e7e7',
+      headerActionButtonColor: '#4d4d4d'
+    }
+  };
+
+  function camelVariable(variable) {
+    return variable.replace(
+      /-([\w-])/g,
+      (_, p1) => p1 === '-' ? '' : p1.toUpperCase()
+    );
+  }
+
+  function getPaletteWithKeywords(arrayPalette) {
```

Comment:
from an array of colors, we get an object with the keywords as keys and the colors as values.